### PR TITLE
Upgrade Ruby backend to Beta Client Libraries

### PR DESCRIPTION
**Pull request by lindawoo-plaid**
Updates Ruby backend from 11.0.0 API Client Libraries to the 14.0.0.beta.2 Beta version.

### CODE REVIEW

File: ruby/app.rb [hunk 1]
```diff
@@ -176,36 +204,60 @@
...
+        raise e if json_response['error_code'] != 'PRODUCT_NOT_READY'
+
+        sleep 1
       end
-      error_response = format_error(e)
-      pretty_print_response(error_response)
-      content_type :json
-      return error_response.to_json
     end
+  rescue Plaid::ApiError => e
```
[the remark this answers is not in the document]
Comment (lindawoo-plaid) [reply ruby/app.rb:256]: got it...deleted the second rescue

File: ruby/.bundle/config [hunk 2]
```diff
@@ -0,0 +1,2 @@
+---
+BUNDLE_BIN: "bin"
```
Comment (skylarmb) [inline ruby/.bundle/config:2]: the `.bundle` folder should not be committed

File: ruby/app.rb [hunk 3]
```diff
@@ -7,17 +7,25 @@
 require 'json'
 require 'plaid'
 require 'sinatra'
+require 'pp'
```
Comment (skylarmb) [inline ruby/app.rb:10]: what is this?

File: ruby/app.rb [hunk 4]
```diff
@@ -176,36 +204,56 @@
...
+      user: {
+        client_user_id: '789',
+        first_name: 'Jane',
+        middle_name: 'Leah',
+        last_name: 'Doe',
+        ssn: '123-45-6789',
+        phone_number: '(555) 123-4567',
+        email: 'jane.doe@example.com'
+      }
+    }
```
Comment (skylarmb) [inline ruby/app.rb:219]: why was this not required in the old code?

File: ruby/app.rb [hunk 5]
```diff
@@ -176,36 +204,56 @@
...
+      rescue Plaid::ApiError => e
+        json_response = JSON.parse(e.response_body)
+        raise e if json_response['error_code'] != 'PRODUCT_NOT_READY'
+
+        sleep 1
       end
-      error_response = format_error(e)
-      pretty_print_response(error_response)
-      content_type :json
-      return error_response.to_json
```
Comment (skylarmb) [inline ruby/app.rb:207]: the new code seems substantially different than the old code, could you explain the change a little? Where did the pretty printing the error go? Where fo we return the error now?

File: ruby/app.rb [hunk 6]
```diff
@@ -256,23 +318,22 @@
...
-      redirect_uri: nil_if_empty_envvar('PLAID_REDIRECT_URI'),
+    link_token_create_request = Plaid::LinkTokenCreateRequest.new(
+      {
+        :user => { :client_user_id => 'user-id' },
+        :client_name => 'Plaid Quickstart',
+        :products => ENV['PLAID_PRODUCTS'].split(','),
+        :country_codes => ENV['PLAID_COUNTRY_CODES'].split(','),
+        :language => "en",
+        :redirect_uri => nil_if_empty_envvar('PLAID_REDIRECT_URI')
+      }
```
Comment (skylarmb) [inline ruby/app.rb:329]: it seems we are mixing syntax here, sometimes we are using `...Request.new({ :foo => bar })` and other times we are using `Request.new` and then setting each attribute one by one. Could we choose just one? I personally prefer this syntax over setting each attribute on a new line. What do you think?

Comment (skylarmb) [inline ruby/app.rb:329]: Also does this syntax work? I think its equivalent but cleaner
```suggestion
    link_token_create_request = Plaid::LinkTokenCreateRequest.new(
      {
        user: { client_user_id: 'user-id' },
        client_name: 'Plaid Quickstart',
        products: ENV['PLAID_PRODUCTS'].split(','),
        country_codes: ENV['PLAID_COUNTRY_CODES'].split(','),
        language: "en",
        redirect_uri: nil_if_empty_envvar('PLAID_REDIRECT_URI')
      }
```

File: ruby/app.rb [hunk 7]
```diff
@@ -341,15 +416,37 @@ def nil_if_empty_envvar(field)
...
+      error_type: body['error_type']
     }
   }
 end
 
 def pretty_print_response(response)
   puts JSON.pretty_generate(response)
 end
+
+def poll_for_asset_report(asset_report_token)
```
Comment (skylarmb) [inline ruby/app.rb:434]: is this function used?

**Review by skylarmb — CHANGES_REQUESTED** [review]

File: ruby/app.rb (hunk 4 above)
Comment (lindawoo-plaid) [reply ruby/app.rb:219]: none of the options are necessarily required.  This was added in the Ruby test in the generated clibs, so I left it in.

File: ruby/app.rb (hunk 3 above)
Comment (lindawoo-plaid) [reply ruby/app.rb:10]: this had to do with pretty pretty print.  I don't think it is necessary anymore. I will delete.

**Review by skylarmb — DISMISSED** [review]
LGTM other than the .gitignore which will create a merge conflict once one of these is merged. Nice work!!!

File: ruby/.gitignore [hunk 8]
```diff
@@ -0,0 +1,6 @@
+# vs code
+ruby/.vscode/launch.json
+
+#ruby
```
Comment (skylarmb) [inline ruby/.gitignore:4]:
```suggestion
```

**Review by skylarmb — CHANGES_REQUESTED** [review]

**Review by skylarmb — APPROVED** [review]

**Pull request merged**